apollo-router-core/src/plugin.rs

```diff
@@ -64,49 +64,71 @@ pub fn plugins() -> HashMap<String, PluginFactory> {
     PLUGIN_REGISTRY.lock().expect("Lock poisoned").clone()
 }
 
+/// All router plugins must implement the Plugin trait. This trait defines lifecycle hooks that enable hooking into Apollo Router services.
+/// The trait also provides a default implementations for each hook, which returns the associated service unmodified.
+/// For more information about the plugin lifecycle please check this documentation https://www.apollographql.com/docs/router/customizations/native/#plugin-lifecycle
 #[async_trait]
 pub trait Plugin: Send + Sync + 'static + Sized {
     type Config: JsonSchema + DeserializeOwned;
 
+    /// This is invoked once after the router starts and compiled-in
+    /// plugins are registered.
     fn new(config: Self::Config) -> Result<Self, BoxError>;
 
-    // Plugins will receive a notification that they should start up and shut down.
+    /// Plugins will receive a notification that they should start up and shut down.
+    /// This is invoked whenever configuration is changed
+    /// during router execution, including at startup.
     async fn startup(&mut self) -> Result<(), BoxError> {
         Ok(())
     }
+
+    /// This is invoked whenever configuration is changed
+    /// during router execution, including at shutdown.
     async fn shutdown(&mut self) -> Result<(), BoxError> {
         Ok(())
     }
 
+    /// This service runs at the very beginning and very end of the request lifecycle.
```

Comment:
maybe beginning and end is not clear here, it should emphasize that it runs when receiving the request and before sending the response

Reply:
Added `It's the entrypoint of every requests and also the last hook before sending the response.`